ngraph/frontend/onnx_editor/src/edge_mapper.cpp

```diff
@@ -0,0 +1,218 @@
+                           "is out of scope inputs list");
+    }
+    if (input_index >= m_node_inputs[node_index].size())
+    {
+        throw ngraph_error("Node with index: " + std::to_string(node_index) +
+                           " has not input with index: " + std::to_string(input_index));
+    }
+    const auto input_name = m_node_inputs[node_index][input_index];
+    return input_name;
+}
+
+InputEdge onnx_editor::EdgeMapper::to_input_edge(const Node& node, const Input& in) const
+{
+    // identification can be both based on node name and output name
+    const auto& node_indexes = find_node_indexes(node.m_node_name, node.m_output_name);
+    int node_index = -1;
+    if (node_indexes.size() == 1)
+    {
+        node_index = node_indexes[0];
+    }
+    else if (!in.m_input_name.empty()) // input indexes are not deterministic
+    {
+        // many nodes with the same name
+        // check if some of found index matches input name
+        int matched_inputs_number = 0;
+        for (const auto& index : node_indexes)
+        {
+            if (std::count(std::begin(m_node_inputs[index]),
+                           std::end(m_node_inputs[index]),
+                           in.m_input_name) > 0)
+            {
+                node_index = index;
+                ++matched_inputs_number;
+            }
+        }
+        if (matched_inputs_number == 0)
+        {
+            throw ngraph_error("Input edge described by: " + node.m_node_name +
+                               " and input name: " + in.m_input_name + " was not found");
+        }
+        if (matched_inputs_number > 1)
+        {
+            throw ngraph_error("Given node name: " + node.m_node_name + " and input name: " +
+                               in.m_input_name + " are ambiguous to determine input edge");
+        }
+    }
+    else
+    {
+        throw ngraph_error("Given node name: " + node.m_node_name +
+                           " and input index: " + std::to_string(in.m_input_index) +
+                           " are ambiguous to determine input edge");
+    }
+    if (!in.m_input_name.empty())
+    {
+        return InputEdge{node_index, in.m_input_name};
+    }
+    else if (in.m_input_index != -1) // input index is set
+    {
+        const auto& input_name = get_node_input_name(node_index, in.m_input_index);
+        return InputEdge{node_index, input_name};
+    }
+    else
+    {
+        throw ngraph_error("Not enough information to determine input edge");
+    }
+}
+
+OutputEdge onnx_editor::EdgeMapper::to_output_edge(const Node& node, const Output& out) const
+{
+    // identification can be both based on node name and output name
+    const auto& node_indexes = find_node_indexes(node.m_node_name, node.m_output_name);
+    int node_index = -1;
+    if (node_indexes.size() == 1)
+    {
+        node_index = node_indexes[0];
+    }
+    else if (!out.m_output_name.empty()) // output indexes are not deterministic
```

Comment:
I thought that in onnx model order of inputs and outputs is fixed.

Reply:
The comment has been not precised. It should be `// output indexes are not deterministic if node name is ambiguous`